### ISSUE TITLE
Layer/Startup Overhead Question

### ISSUE THREAD

**Issue by justinrcs** [newrelic/node-newrelic#314]
We have the New Relic Node agent implemented via Lambda Layers.

We're seeing 6-7 second startup time for the cold-start layer initialization for our Lambda functions when the layer is applied.  From CloudWatch logs:
 
> 08:58:15 Cross application tracing is explicitly disabled in serverless_mode.
> 08:58:15 The native-metrics module is disabled by default when serverless_mode is enabled. If desired, enable the native-metrics module via config file or environment variable.
> 08:58:15 Using New Relic for Node.js. Agent version: 6.1.0; Node version: v10.17.0.
> 08:58:15 Agent state changed from stopped to starting.
> 08:58:15 New Relic for Node.js starting in serverless mode -- skipping connection process
> 08:58:15 START RequestId: 29b543f9-6c19-416c-a30a-a4e2267be9a1 Version: $LATEST
> **08:58:21** entering RaptorContentManagementAPI-dev-initializeDownload

Compare this to the same function when the layer is not applied:

> 19:57:53 START RequestId: 28b48b3d-e44c-40a1-be9e-a4ef326d621e Version: $LATEST
> 19:57:53 entering RaptorContentManagementAPI-dev-initializeDownload

Memory for this function is set to: 320 MB, prior to applying the layer it was 256 MB

Questions:
1. Is this expected behavior?
2. Are there any tuning parameters we can use?
3. What is the recommended memory requirements for instrumentation?

**Comment by astormnewrelic** [comment]
@justinrcs Nice to meet you and thanks for bringing this to our attention.  

First (with advanced apologies for the ping-pong), the best/fastest way to get support for a New Relic product is via our official support channels: https://support.newrelic.com/  Reporting these sorts of issues through official channels is the fastest way to get an issue in front of the right people, and get the most up-to-date steps for debugging the issues.

That said we're always happy to take a look and see what we can do here.  

That sort of cold start time is not expected behavior.   Without knowing what the lambda doing it's hard to say what's going on. 

Seeing trace level logs during a cold start might reveal more about what the agent's doing during  those six seconds.  

Also, can you talk a bit about what your lambda function is doing (and if it's open-source or source-available somewhere share it)?   The TL;DR; on this is your lambda function is doing something that the agent's inefficiently instrumenting.  Step one to solving this is figuring out what that something is.

**Comment by justinrcs** [comment]
Thanks - appreciate the follow-up. I have opened a support ticket.

We're noticing that the 6-7 second startup time is consistent across the many functions that we have deployed with the layer.

This specific Lambda function looks up the appropriate AWS region and generates a presigned URL to a piece of content stored in a private S3 bucket. 

I will enable tracing on the agent to see if I can get more information about what it's doing.
